server/stream.go

```diff
@@ -1846,7 +1846,7 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 							var err error
 							si.trs[i], err = NewSubjectTransform(s.SubjectTransforms[i].Source, s.SubjectTransforms[i].Destination)
 							if err != nil {
-								mset.srv.Errorf("Unable to get subject transform for source: %v", err)
+								return fmt.Errorf("unable to get subject transform for source: %v", err)
```

Comment:
You will need to unlock here before returning I would imagine.

Reply:
Yes, that's the reason the test was hanging on the server shutdown. Added it (back, since it was removed in #5571, we're getting there in the end :)).